proposals/imageverificationpolicy.md

```diff
@@ -0,0 +1,289 @@
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+  - [Goals:](#goals)
+- [Proposal](#proposal)
+  - [Match Constraints](#match-constraints)
+  - [Images](#images)
+  - [Attestors](#attestors)
+  - [Attestations](#attestations)
+  - [Verifications](#verifications)
+    - [Custom CEL functions](#custom-cel-functions)
+      - [VerifyImageSignatures](#verifyimagesignatures)
+      - [VerifyAttestationSignatures](#verifyattestationsignatures)
+    - [Payload](#payload)
+    - [ParseImageReference](#parseimagereference)
+  - [Other details](#other-details)
+    - [Caching](#caching)
+    - [Digest Mutation](#digest-mutation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Prior Art](#prior-art)
+- [Unresolved Questions](#unresolved-questions)
+
+# Overview
+[overview]: #overview
+
+This document proposes the new design for image verification policies. 
+
+# Definitions
+[definitions]: #definitions
+
+1. **Image:** Images are static files that contain instructions for building a container and running an isolated process on a containerization platform.
+2. **Metadata:** Metadata is used to describe software and the build environment. Provenance (origin) data, SBOMs, and vulnerability scan reports are the essential set of metadata required to assess security risks for software.
+3. **Attestation:**  Authenticated metadata is used to attest to the integrity of a software system. Both custom and standardized metadata can be converted into attestations.
+4. **Attester:** An identity, such as a key or certificate that confirms or verifies the authenticity of an image or an attestors
+
+# Motivation
+[motivation]: #motivation
+
+Image verification policies were added to Kyverno in v1.4, and it has since added support for notary as well as attestations verification. The incremental growth of verify images policies has led to a complex structure and a confusing syntax. 
+Kubernetes has changed a lot in the last few years and has added direct support for CEL but Kyverno's verifyImages policies still use JSON patch and JMESPath.
+
+This redesign will help in simplifying the image verification policy structure and also integrate CEL into image verification.
+
+## Goals:
+- Simplify the syntax of image verification policy
+- Integrate CEL into image verification
+- Make image verification independent of Kubernetes
+
+# Proposal
+
+Here is a high level overview of the image verification policy CRD:
+```yaml
+apiVersion: kyverno.io/v2alpha1
+kind: ImageVerificationPolicy
+metadata:
+  name: sample
+spec:
+  failurePolicy: Fail
+  failureAction: Enforce
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["pods"]
+    imageRules:
+      references:
+      - "ghcr.io/*"
+      expressions:
+      - $image == "nginx"
+  images:
```

Comment:
Why not have these as built-in variables? Seems the user needs to repeat this declaration in all policies?

Reply:
images in images section will be prefetched and cached